### DISCUSSION TITLE
How to get new ADE key for Calibre deDRM 10.0.3

### DISCUSSION THREAD

**Discussion by McKnitter** [apprenticeharper/DeDRM_tools#2262] [Q&A] [answered]
I stupidly customised the Calibre deDRM plug-in and deleted the ADE key. I was uninstalling Adobe Digital Editions 3.0 and installing the older ADE 2.0 so I could still transfer some older epubs into Calibre.

Both Calibre 4.23 and ADE 2.0 are on the same laptop. I'm sure my removing the ADE key stuffed it up. 

I can download epubs into ADE 2.0 (via the acsm links), and I can read them there. But they are all DRM locked when I add them to Calibre. I can add a 'free' epub from ADE to Calibre, of course.

Is it fixable?

**Comment by ElleKayEm** [comment, marked as answer]
For calibre 4.23, I'd try DeDRM 6.8.1: https://github.com/apprenticeharper/DeDRM_tools/releases/tag/v6.8.1

**Reply by ElleKayEm** [reply]
Possible 10.0.9 might work: https://github.com/noDRM/DeDRM_tools/releases/tag/v10.0.9

**Reply by McKnitter** [reply]
I'll give them each a try - thanks!

**Reply by McKnitter** [reply]
6.8.1 worked! The key is back and I successfully imported a book I had imported before I 'lost' the key. I had attempted to add it again using the 10.0.3 plugin, and it hadn't worked. I think I'll just stick with 6.8.1 for now. 

10.0.3 worked when I was using ADE 3.0, but I want to try an old version of ADE for a while. THANKS!

**Reply by ElleKayEm** [reply]
DeDRM 10.0.3 does have a bug that causes problems if you try to retrieve the Adobe Key from within DeDRM customization.  The trick is to import an ADE book and let DeDRM retrieve the key automatically.  That should be fixed in 10.0.9.  But since you're using older calibre and older ADE, might as well stick with the older DeDRM that's known to work with them.

**Reply by McKnitter** [reply]
I'm sure my problem was my own fault. In the cusomization, I actually removed, with the red X, the Adobe key, thinking I would need a different one (duh) when I went back to ADE 2.0.  But after I installed 2.0 and then tried to import a book to Calibre, DeDRM wouldn't pick up the key. I just kept getting blank covers and "DRM locked" error messages. My fault entirely for fiddling. I am old enough to know better.
THANK YOU 😊

**Reply by qdesjo** [reply]
Thank you so munch for your tips ! 

I was struck for hours and it worked with the kobo pluging in this version.
